pkgs/build-support/fetchzip/default.nix

```diff
@@ -75,16 +67,16 @@ fetchurl ((
       mv "$unpackDir/$fn" "$out"
     '' else ''
       mv "$unpackDir" "$out"
-    '')
-    + ''
+    '') + ''
       ${postFetch}
-    '' + ''
       ${extraPostFetch}
     ''
-
     # Remove non-owner write permissions
     # Fixes https://github.com/NixOS/nixpkgs/issues/38649
+    # (This line should not be merged with the above, because the
+    # above actully evaluates to an empty string most of the time,
+    # making things more efficient.)
```

Comment:
```suggestion
    # and Nix's string concatenation has a fast path when one of
    # the arguments is the empty string.)
```

Reply:
... although the eval performance report shows only ~0.1% improvement on any metric other than cpu time (which is a very noisy metric)...